src/adaptive_resortable_list.h

```diff
@@ -32,89 +31,143 @@
+    size_t expected;    ///< Number of expected keywords.
+    size_t wanted;      ///< Number of wanted keywords
+                        ///< i.e. the number of keywords found and expected
 
-    size_t relinkings;  // the number of relinkings we have made so far
+    size_t relinkings;  ///< Number of relinkings made so far.
 
-    size_t allocated;   // the number of keywords allocated
-    size_t fred;        // the number of keywords cleaned up
+    size_t allocated;   ///< Number of keyword allocations made.
+    size_t fred;        ///< Number of keyword cleaneups made.
 
-    size_t rechecks;    // the number of iterations between re-checks of the
-                        // wanted number of keywords
-                        // this is only needed in cases where the source
-                        // is having less lines over time.
+    size_t rechecks;    ///< Number of iterations between re-checks of the
+                        ///< wanted number of keywords.
+                        ///< This is only needed in cases where the source
+                        ///< is having less lines over time.
 
-    size_t added;       // it is non-zero if new keywords have been added
-                        // this is only needed to detect new lines have
-                        // been added to the file, over time.
+    size_t added;       ///< Number of new keywords in current iteration.
+                        ///< This is only needed to detect new lines have
+                        ///< been added to the file, over time.
 
 #ifdef NETDATA_INTERNAL_CHECKS
-    size_t fast;        // the number of times we have taken the fast path
-    size_t slow;        // the number of times we have taken the slow path
+    size_t fast;        ///< Number of times we have taken the fast path.
+    size_t slow;        ///< Number of times we have taken the slow path.
 #endif
 
-    // the processor to do the job
+    /**
+     * Callback to update `dst` with `value` if we found an expected name.
+     *
+     * @param name The matching keyword.
+     * @param hash of `name`
+     * @param value of `name` 
+     * @param dst to copy `value` into
+     */
     void (*processor)(const char *name, uint32_t hash, const char *value, void *dst);
 
-    // the linked list of the keywords
+    /// Linked list of the keywords
     ARL_ENTRY *head;
 
-    // since we keep the list of keywords sorted (as found in the source data)
-    // this is next keyword that we expect to find in the source data.
+    /// The next keyword we expect to find in the source data.
     ARL_ENTRY *next_keyword;
 } ARL_BASE;
 
-// create a new ARL
+/**
+ * Create a new ARL.
+ *
+ * If `processor` is `NULL` the default processor is used.
+ * It converts the value to an unsigned long long.
+ * \dontinclude adaptive_resortable_list.c
+ * \skip arl_callback_str2ull
+ * \until }
+ *
+ * \todo How to choose a value for rechecks?
```

Comment:
Let's assume a file has 10 keywords. We are only interested for the first 3. ARL will stop parsing the file when all 3 are satisfied (i.e. it will not even read the 4th keyword).

After some time, the file has 9 keywords and one of the wanted ones is missing. ARL needs 3, it does not find them, so it parses the whole file again and again.

`rechecks` provides the number of iterations, after which ARL will re-examine the situation and decide what is best to do. It does not do this all the time, since it requires a complete walk through of the ARL keywords.

So, with `rechecks = 2`, ARL will do this complete walk through every 2 iterations and decide on the number of wanted keywords.

The same could happen for the opposite condition. The file has 10 keywords, we are interested for 4, but only 3 are in the file, then after some time the 4th keyword is added. ARL may not "see" the added keyword, for `recheck` number of iterations.

Too small `recheck` and CPU resources are wasted.
Too big `recheck` and information may be missing from the charts.